### PR TITLE
[feature] Upgrade input & form

### PR DESCRIPTION
**Pull request by baiyaaaaa**
Please make sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer relative issues for you PR.

1. form item support success validate status
2. input support suffix & preffix slot

### CODE REVIEW

**Comment by baiyaaaaa** [comment]
缺英文翻译

**Comment by Leopoldthecoder** [comment]
测试挂了

File: examples/docs/en-US/cascader.md [hunk 1]
```diff
@@ -302,6 +302,7 @@ There are two ways to expand child option items.
     expand-trigger="hover"
     :options="options"
     v-model="selectedOptions2"
+    clearable
```
Comment (Leopoldthecoder) [inline examples/docs/en-US/cascader.md]: 去掉吧

File: packages/theme-chalk/src/input.scss [hunk 2]
```diff
@@ -38,15 +38,36 @@
...
     text-align: center;
     color: $--input-icon-color;
     transition: all .3s;
+  }
+
+  @include e(icon) {
+    height: 100%;
+    width: 25px;
+    text-align: center;
+    // color: $--input-icon-color;
```
Comment (Leopoldthecoder) [inline packages/theme-chalk/src/input.scss]: 去掉吧

File: packages/theme-chalk/src/select.scss [hunk 3]
```diff
@@ -27,27 +27,26 @@
...
 
       @include when(show-close) {
         transition: 0s;
-        height: 16px;
-        width: 16px;
+        // height: 16px;
+        // width: 16px;
         font-size: $--select-font-size;
-        right: 8px;
+        // right: 8px;
```
Comment (Leopoldthecoder) [inline packages/theme-chalk/src/select.scss]: 去掉吧

File: packages/theme-default/src/cascader.css [hunk 4]
```diff
@@ -29,6 +29,16 @@
...
 
+    .el-input__suffix-inner {
+      position: relative;
+      z-index: 2;
+    }
+
+    @e clearIcon {
+      z-index: 2;
+      position: relative;
+    }
```
Comment (Leopoldthecoder) [inline packages/theme-default/src/cascader.css]: 去掉吧

File: test/unit/index.js [hunk 5]
```diff
@@ -4,7 +4,7 @@ Function.prototype.bind = require('function-bind');
 require('packages/theme-default/src/index.css');
 
 // require all test files (files that ends with .spec.js)
-const testsContext = require.context('./specs', true, /\.spec$/);
+const testsContext = require.context('./specs', true, /cascader\.spec$/);
```
Comment (Leopoldthecoder) [inline test/unit/index.js]: 🌚

**Pull request merged**